spring-boot-project/spring-boot/src/test/java/org/springframework/boot/web/client/RestTemplateBuilderTests.java

```diff
@@ -497,6 +499,47 @@ public void unwrappingDoesNotAffectRequestFactoryThatIsSetOnTheBuiltTemplate() {
 				.isInstanceOf(BufferingClientHttpRequestFactory.class);
 	}
 
+	@Test
+	@SuppressWarnings("deprecation")
+	public void deprecatedReadTimeout() {
+		ClientHttpRequestFactory requestFactory = this.builder
+				.requestFactory(SimpleClientHttpRequestFactory.class).setReadTimeout(1234)
+				.build().getRequestFactory();
+		assertThat(ReflectionTestUtils.getField(requestFactory, "readTimeout"))
+				.isEqualTo(1234);
+	}
+
+	@Test
+	@SuppressWarnings("deprecation")
+	public void deprecatedConnectTimeout() {
+		ClientHttpRequestFactory requestFactory = this.builder
+				.requestFactory(SimpleClientHttpRequestFactory.class)
+				.setConnectTimeout(1234).build().getRequestFactory();
+		assertThat(ReflectionTestUtils.getField(requestFactory, "connectTimeout"))
+				.isEqualTo(1234);
+	}
+
+	@Test
+	@SuppressWarnings("deprecation")
+	public void nullReadTimeoutShouldNotAppliedIgnorePreviousSettings() {
```

Comment:
Those two tests are new right? You could do the same with one that takes a `Duration` and one that takes `null` afteR. No need for the deprecation. Besides, I'd appreciate a separate PR for this as they are unrelated.